plugins/woocommerce/src/Blocks/BlockTypes/ProductImage.php

```diff
@@ -71,6 +71,18 @@ private function render_on_sale_badge( $product, $attributes ) {
 
 		$font_size = StyleAttributesUtils::get_font_size_class_and_style( $attributes );
 
+		/**
+		 * Filters the product sale badge text.
+		 *
+		 * @hook woocommerce_sale_badge_text
+		 * @since 9.9.0
+		 *
+		 * @param string $sale_text The sale badge text.
+		 * @param WC_Product $product The product object.
+		 * @return string The filtered sale badge text.
+		 */
+		$sale_text = wp_strip_all_tags( apply_filters( 'woocommerce_sale_badge_text', __( 'Sale', 'woocommerce' ), $product ) );
```

Comment:
Personally, I'm not sure we should wrap this in `wp_strip_all_tags`. We have the `esc_html` call further down to sanitize the string; beyond that, I'm not sure we should make any assumptions about what a third-party developer might return from the filter.